[Bug] ClubChatRoom 삭제 시 TransientObjectException 발생

## ❗ 버그 설명
Club ↔ ClubChatRoom 1:1 연관관계에서 flush 시점에 TransientObjectException 예외가 발생합니다.

## 🔁 재현 방법
1. 클럽 생성 API 호출 (예: POST /api/clubs)
2. 서비스 내부에서 ClubChatRoom 생성 직후(아직 save 전) 클럽/채팅방 삭제 API 호출 시 club_chat_room 삭제 전에 역방향을 끊지 않음
3. 이후 다른 리포지토리 호출/벌크 쿼리 실행으로 auto-flush 트리거

## 🖼️ 스크린샷 또는 로그
~~~ bash
Caused by: org.hibernate.TransientObjectException: persistent instance references an unsaved transient instance of 'com.mobble.mobbleserver.domain.chat.clubChatRoom.entity.ClubChatRoom' (save the transient instance before flushing)
~~~

## ✅ 예상되는 해결 방향 (선택)
- 연관 관리 메서드 도입(소유측 주도)
    - ClubChatRoom.attachTo(Club) : save 후 역방향만 동기화
    - ClubChatRoom.detach() : 파라미터 없이 양쪽 참조 해제(역방향 null → 소유측 null)
    - Club.setClubChatRoomInternal(ClubChatRoom) : null 허용(연관 해제 목적)
- 생성/삭제 순서 표준화
    - 생성: ChatRoom → ChatRoomParticipant → ClubChatRoom(save) → clubChatRoom.attachTo(club)
    - 삭제: clubChatRoom.detach() 먼저 → 참가자/메시지/ClubChatRoom/ChatRoom 삭제

## 💻 환경 정보
- OS / 브라우저: macOS
- 요청 경로(URL): POST /api/clubs, DELETE /api/clubs/{clubId} 등 연관 생성/삭제 엔드포인트
- 기타 환경 정보:

--- a/src/main/java/com/mobble/mobbleserver/domain/chat/clubChatRoom/entity/ClubChatRoom.java
+++ b/src/main/java/com/mobble/mobbleserver/domain/chat/clubChatRoom/entity/ClubChatRoom.java
@@ -38,4 +38,17 @@ public class ClubChatRoom {
                 .chatRoom(chatRoom)
                 .build();
     }
+
+    public void attachTo(Club club) {
+        if (this.club != club) throw new IllegalArgumentException("");
+
+        club.setClubChatRoomInternal(this);
+    }
+
+    public void detach() {
+        if (this.club == null) return;
+
+        this.club.setClubChatRoomInternal(null);
+        this.club = null;
+    }
 }

--- a/src/main/java/com/mobble/mobbleserver/domain/chat/clubChatRoom/repository/ClubChatRoomRepository.java
+++ b/src/main/java/com/mobble/mobbleserver/domain/chat/clubChatRoom/repository/ClubChatRoomRepository.java
@@ -3,7 +3,11 @@ package com.mobble.mobbleserver.domain.chat.clubChatRoom.repository;
 import com.mobble.mobbleserver.domain.chat.clubChatRoom.entity.ClubChatRoom;
 import org.springframework.data.jpa.repository.JpaRepository;
 
+import java.util.Optional;
+
 public interface ClubChatRoomRepository extends JpaRepository<ClubChatRoom, Long> {
 
     boolean existsClubChatRoomByClubId(Long clubId);
+
+    Optional<ClubChatRoom> findByClubId(Long clubId);
 }

--- a/src/main/java/com/mobble/mobbleserver/domain/chat/clubChatRoom/service/ClubChatRoomService.java
+++ b/src/main/java/com/mobble/mobbleserver/domain/chat/clubChatRoom/service/ClubChatRoomService.java
@@ -41,14 +41,14 @@ public class ClubChatRoomService {
 
     private final ChatMessageService chatMessageService;
 
-    private final MemberValidator memberValidator;
-    private final ClubMemberValidator clubMemberValidator;
-    private final ClubChatRoomValidator clubChatRoomValidator;
-
     private final ChatRoomParticipantRepository chatRoomParticipantRepository;
     private final ChatRoomRepository chatRoomRepository;
     private final ChatMessageRepository chatMessageRepository;
     private final ClubChatRoomRepository clubChatRoomRepository;
+
+    private final MemberValidator memberValidator;
+    private final ClubMemberValidator clubMemberValidator;
+    private final ClubChatRoomValidator clubChatRoomValidator;
 
     @Transactional
     public void sendGroupMessage(ClubChatMessageRequestDto dto, Long memberId) {
@@ -67,16 +67,23 @@ public class ClubChatRoomService {
     }
 
     @Transactional
-    public ClubChatRoomPreviewResponseDto createClubChatRoom(Club club, Member member) {
+    public ClubChatRoomPreviewResponseDto createClubChatRoom(Long clubId, Long memberId) {
+        ClubMember clubMember = clubMemberValidator.findClubMemberByClubIdAndMemberIdOrThrow(clubId, memberId);
+        Club club = clubMember.getClub();
+        Member member = clubMember.getMember();
+
         clubChatRoomValidator.existsClubChatRoomByClubIdOrThrow(club.getId());
 
         ChatRoom chatRoom = ChatRoom.createChatRoom(ChatRoomType.GROUP);
-        ChatRoomParticipant participant = ChatRoomParticipant.createChatRoomParticipant(chatRoom, member);
-        ClubChatRoom clubChatRoom = ClubChatRoom.createClubChatRoom(club, chatRoom);
-
         chatRoomRepository.save(chatRoom);
+
+        ChatRoomParticipant participant = ChatRoomParticipant.createChatRoomParticipant(chatRoom, member);
         chatRoomParticipantRepository.save(participant);
+
+        ClubChatRoom clubChatRoom = ClubChatRoom.createClubChatRoom(club, chatRoom);
         clubChatRoomRepository.save(clubChatRoom);
+
+        clubChatRoom.attachTo(club);
 
         return ClubChatRoomPreviewResponseDto.toDto(chatRoom, club, null, 0, null);
     }
@@ -136,10 +143,11 @@ public class ClubChatRoomService {
     }
 
     @Transactional
-    public void deleteClubChatRoom(Club club) {
-        ClubChatRoom clubChatRoom = club.getClubChatRoom();
-        if (clubChatRoom == null) return;
+    public void deleteClubChatRoom(Long clubId) {
+        ClubChatRoom clubChatRoom = clubChatRoomValidator.findClubChatRoomByClubIdOrThrow(clubId);
         ChatRoom chatRoom = clubChatRoom.getChatRoom();
+
+        clubChatRoom.detach();
 
         chatRoomParticipantRepository.deleteByChatRoomId(chatRoom.getId());
         chatMessageRepository.deleteByChatRoomId(chatRoom.getId());

--- a/src/main/java/com/mobble/mobbleserver/domain/chat/clubChatRoom/validator/ClubChatRoomValidator.java
+++ b/src/main/java/com/mobble/mobbleserver/domain/chat/clubChatRoom/validator/ClubChatRoomValidator.java
@@ -1,5 +1,6 @@
 package com.mobble.mobbleserver.domain.chat.clubChatRoom.validator;
 
+import com.mobble.mobbleserver.domain.chat.clubChatRoom.entity.ClubChatRoom;
 import com.mobble.mobbleserver.domain.chat.clubChatRoom.repository.ClubChatRoomRepository;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
@@ -12,5 +13,10 @@ public class ClubChatRoomValidator {
 
     public void existsClubChatRoomByClubIdOrThrow(Long clubId) {
         if (clubChatRoomRepository.existsClubChatRoomByClubId(clubId)) throw new IllegalArgumentException("");
+    }
+
+    public ClubChatRoom findClubChatRoomByClubIdOrThrow(Long clubId) {
+        return clubChatRoomRepository.findByClubId(clubId)
+                .orElseThrow(() -> new IllegalArgumentException(""));
     }
 }

--- a/src/main/java/com/mobble/mobbleserver/domain/club/core/entity/Club.java
+++ b/src/main/java/com/mobble/mobbleserver/domain/club/core/entity/Club.java
@@ -83,6 +83,10 @@ public class Club extends BaseEntity {
                 .build();
     }
 
+    public void setClubChatRoomInternal(ClubChatRoom chatRoom) {
+        this.clubChatRoom = chatRoom;
+    }
+
     public void updateClub(
             ClubCategory category,
             String name,

--- a/src/main/java/com/mobble/mobbleserver/domain/club/core/service/ClubService.java
+++ b/src/main/java/com/mobble/mobbleserver/domain/club/core/service/ClubService.java
@@ -1,6 +1,8 @@
 package com.mobble.mobbleserver.domain.club.core.service;
 
 import com.mobble.mobbleserver.domain.article.repository.ArticleRepository;
+import com.mobble.mobbleserver.domain.chat.clubChatRoom.dto.response.ClubChatRoomPreviewResponseDto;
+import com.mobble.mobbleserver.domain.chat.clubChatRoom.service.ClubChatRoomService;
 import com.mobble.mobbleserver.domain.club.ageGroup.entity.AgeGroup;
 import com.mobble.mobbleserver.domain.club.ageGroup.entity.AgeGroupType;
 import com.mobble.mobbleserver.domain.club.ageGroup.repository.AgeGroupRepository;
@@ -37,6 +39,8 @@ import java.util.List;
 @Transactional(readOnly = true)
 public class ClubService {
 
+    private final ClubChatRoomService clubChatRoomService;
+
     private final ClubRepository clubRepository;
     private final ClubCategoryRepository clubCategoryRepository;
     private final ClubMemberRepository clubMemberRepository;
@@ -64,6 +68,9 @@ public class ClubService {
 
         List<AgeGroup> ageGroups = createClubAgeGroups(club, dto.ageGroup());
         ageGroupRepository.saveAll(ageGroups);
+
+        // Todo: 반환값이 Club 채팅방의 preview 에 필요한 데이터이기 때문에 반환 DTO에 포함 되어야 함
+        ClubChatRoomPreviewResponseDto clubChatRoom = clubChatRoomService.createClubChatRoom(club.getId(), member.getId());
 
         return buildClubResponse(club, member, member.getName());
     }
@@ -98,25 +105,27 @@ public class ClubService {
 
     @Transactional
     public void deleteClub(Long clubId, Long memberId) {
-        Club club = clubValidator.findClubByClubIdOrThrow(clubId);
-        Member member = memberValidator.findMemberByMemberIdOrThrow(memberId);
         ClubMember clubMember = clubMemberValidator.findClubMemberByClubIdAndMemberIdOrThrow(clubId, memberId);
+        Club club = clubMember.getClub();
+
         assertLeader(clubMember);
 
-        clubLikeRepository.deleteClubLikeAllByClub_Id(clubId);
+        clubChatRoomService.deleteClubChatRoom(club.getId());
 
-        List<Long> articleIds = articleRepository.findArticleIdsByClubId(clubId);
+        clubLikeRepository.deleteClubLikeAllByClub_Id(club.getId());
+
+        List<Long> articleIds = articleRepository.findArticleIdsByClubId(club.getId());
 
         commentLikeRepository.deleteAllCommentLikeByComment_Article_IdIn(articleIds);
         commentRepository.deleteAllCommentByArticle_IdIn(articleIds);
         articleLikeRepository.deleteAllArticleLikeByArticle_IdIn(articleIds);
-        articleRepository.deleteAllArticleByClub_Id(clubId);
-        clubMemberRepository.deleteAllClubMemberByClubId(clubId);
+        articleRepository.deleteAllArticleByClub_Id(club.getId());
+        clubMemberRepository.deleteAllClubMemberByClubId(club.getId());
 
-        clubLikeRepository.deleteClubLikeAllByClub_Id(clubId);
-        ageGroupRepository.deleteAllClubAgeGroupByClubId(clubId);
+        clubLikeRepository.deleteClubLikeAllByClub_Id(club.getId());
+        ageGroupRepository.deleteAllClubAgeGroupByClubId(club.getId());
 
-        clubRepository.deleteById(clubId);
+        clubRepository.deleteById(club.getId());
     }
 
     private ClubCategory findCategoryOrThrow(String categoryName) {

